components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -577,12 +583,39 @@ func (g *RTKMovementSensor) Position(ctx context.Context, extra map[string]inter
 	g.ntripMu.Lock()
 	lastError := g.err.Get()
 	if lastError != nil {
-		defer g.ntripMu.Unlock()
-		return geo.NewPoint(0, 0), 0, lastError
+		lastPosition := g.lastposition.GetLastPosition()
+		g.ntripMu.Unlock()
+		if lastPosition != nil {
+			return lastPosition, 0, nil
+		}
+		return nil, 0, lastError
```

Comment:
I believe this will cause a nil pointer error stemming from the underlying library, best to return `math.Nan` here as well and handle that.

Reply:
I believe that.